src/datasources/floodscan.py

```diff
@@ -19,7 +23,240 @@
 )
 
 
+def calculate_recent_flood_exposure_rasters(
+    iso3: str, clobber: bool = False, verbose: bool = False
+):
+    """
+    Calculate recent (2024 onwards) flood exposure rasters for a given country.
+    Parameters
+    ----------
+    iso3: str
+        ISO3 code of the country
+    clobber: bool
+        Whether to overwrite existing data
+    verbose: bool
+        Whether to print progress of specific dates
+
+    Returns
+    -------
+
+    """
+    pop = worldpop.load_worldpop_from_blob(iso3)
+    # check for existing raw Floodscan rasters
+    existing_fs_raw_files = [
+        x
+        for x in blob.list_container_blobs(
+            name_starts_with="raster/cogs/aer_area_300s_",
+            container_name="global",
+        )
+        if x.endswith(".tif")
+    ]
+    # filter to only 2024 onwards
+    recent_fs_raw_files = [
+        x for x in existing_fs_raw_files if "300s_2024" in x
+    ]
+    # check for existing processed exposure rasters
+    existing_exposure_files = blob.list_container_blobs(
+        name_starts_with=f"{blob.PROJECT_PREFIX}/processed/flood_exposure/"
+        f"{iso3}"
+    )
+
+    # stack up relevant raw Floodscan rasters
+    das = []
+    for blob_name in tqdm(recent_fs_raw_files):
+        date_in = datetime.strptime(blob_name.split("/")[-1][14:22], "%Y%m%d")
+        date_str = date_in.strftime("%Y-%m-%d")
+        exposure_blob_name = get_blob_name(
+            iso3, "exposure_raster", date=date_str
+        )
+        if exposure_blob_name in existing_exposure_files and not clobber:
+            if verbose:
+                print(f"already processed for {date_str}, skipping")
+            continue
+        da_in = blob.open_blob_cog(blob_name, container_name="global")
+        long_name = da_in.attrs["long_name"]
+        if long_name == ("SFED", "MFED"):
+            da_in = da_in.isel(band=0)
+        elif long_name == ("MFED", "SFED"):
+            da_in = da_in.isel(band=1)
+        elif long_name == "SFED":
+            da_in = da_in.isel(band=0)
+        else:
+            print(f"unrecognized long_name, skipping {date_in}")
+            continue
+        da_in = da_in.drop_vars("band")
+        da_in["date"] = date_in
+        da_in = da_in.persist()
+        das.append(da_in)
+
+    if not das:
+        if verbose:
+            print("no new floodscan data to process")
+        return
+    ds_recent = xr.concat(das, dim="date")
+    # filter to only pixels with flood extent > 5% to reduce noise
+    ds_recent_filtered = ds_recent.where(ds_recent >= 0.05)
+    # interpolate to Worldpop grid and
+    # multiply by population to get exposure
+    exposure = ds_recent_filtered.interp_like(pop, method="nearest") * pop
```

Comment:
Here is where we are actually calculating the exposure rasters. I am unsure whether it makes sense to only take pixels with flood extent ≥ 5%. I think we lose a fair amount of information this way, and I'm not sure we really benefit from the reducing the noise.

We also may want to think about whether to multiply by the relevant population raster for _that year_. Also, which population raster to use? There are several options on WorldPop (of which we're using the `2020_1km_Aggregated_UNadj`). There is also [GHSL](https://human-settlement.emergency.copernicus.eu/ghs_pop2023.php).

Reply:
I don't have a lot of background in working with the Floodscan data, but I'd lean more towards removing that 5% threshold. I think we're already doing a lot of smoothing, so this seems a little bit over cautious to remove noise. We'd also want to be able to justify why we picked that 5% number, which seems slightly arbitrary.

Reply:
As discussed- we will ditch the 5% and re-run the historical pipelines later #14